NineAnimator/Models/Anime Source/animeunity.it/AnimeUnity+Featured.swift

```diff
@@ -46,7 +49,41 @@ extension NASourceAnimeUnity {
             let decodedResponse = user
             let recentAnimeLinks = try decodedResponse.map {
                 record -> AnimeLink in
-                let link = "https://animeunity.it/anime/"+String(record.id)+"-"+record.slug
+                let link = self.endpointURL.absoluteString + "/anime/"+String(record.id)+"-"+record.slug
+                var animeUrlBuilder = try URLComponents(
+                    url: link.asURL(),
+                    resolvingAgainstBaseURL: true
+                ).tryUnwrap()
+                animeUrlBuilder.queryItems = [
+                    .init(name: "id", value: "1")
+                ]
+                return AnimeLink(
+                    title: record.title,
+                    link: try animeUrlBuilder.url.tryUnwrap(),
+                    image: try record.imageurl.asURL(),
+                    source: self
+                )
+            }
+            return self.requestManager.request(
+                url: self.endpointURL.absoluteString + "/top-anime",
+                handling: .browsing,
+                query: [ "popular": "true" ]
+            ) .responseData
+          .then {
+            responseContent in
+            let dataFeatured = responseContent
+            let utf8TextFeatured = String(data: dataFeatured, encoding: .utf8) ?? String(decoding: dataFeatured, as: UTF8.self)
+            let bowlFeatured = try SwiftSoup.parse(utf8TextFeatured)
+            var encodedFeatured = try bowlFeatured.select("top-anime").attr("animes")
+            encodedFeatured = encodedFeatured.replacingOccurrences(of: "\n", with: "")
+            let data_jsonFeatured = try encodedFeatured.data(using: .utf8).tryUnwrap()
+            let decoderFeatured = JSONDecoder()
+            decoderFeatured.keyDecodingStrategy = .convertFromSnakeCase
+            let userFeatured: Featured = try decoderFeatured.decode(Featured.self, from: data_jsonFeatured)
+            let decodedResponseFeatured = userFeatured.data
+            let FeaturedAnimeLinks = try decodedResponseFeatured.map {
+                record -> AnimeLink in
+                let link = self.endpointURL.absoluteString + "/anime/"+String(record.id)+"-"+record.slug
```

Comment:
This may cause problem in the future. Use `URL(protocolRelativeString: String, relativeTo baseUrl: URL)` or `self.endpointURL.appendingPathComponent()` instead.

Reply:
```suggestion
                let link = self.endpointURL.appendingPathComponent("anime/" + String(record.id) + "-" + record.slug)
```
in this way? I have to modify also line 54 in the same way?

Reply:
Yep, that looks fine.